src/elements/e-video.vue

```diff
@@ -210,42 +216,52 @@
 
     methods: {
       /**
-       * This is a helper function to validate the videoUrl property
+       * This is a helper for replacing part of a string.
+       * @param original string
+       * @param replace string | null
        */
-      validateUrl(): boolean {
-        if (
-          this.videoUrl
-          && !this.videoUrl.includes(videoUrlSources[this.source])
-        ) {
-          console.error('Invalid video url');
-
-          return false;
+      replaceIdInUrl(
+        original: string,
+        replace: string | null
+      ): string | undefined {
+        if (!replace) {
+          return undefined;
         }
 
-        return true;
+        return original.replace('{videoId}', replace);
+      },
+
+      /**
+       * Helper to check wether a string matches with entries in an array.
+       * @param arr string[]
+       * @param searchString string
+       */
+      arrayHasString(arr: string[], searchString: string): boolean {
+        return arr.some((videoSource) => {
+          if (!searchString.startsWith(videoSource)) {
+            // eslint-disable-next-line no-console
+            console.error('Invalid video url', this.$el);
+
+            return false;
+          }
+
+          return true;
```

Comment:
suggestion: I think the whole evaluation could be shortened to `return arr.some(searchString.startsWith);`.